doc/BGP/BGP-supress-fib-pending.md

```diff
@@ -0,0 +1,804 @@
+  activate RouteOrch
+
+  RouteOrch ->> RouteOrch: doTask()
+
+  loop For each entry in m_toSync
+    RouteOrch ->> RouteBulker: RouteBulker.remove_entry()
+    activate RouteBulker
+    RouteBulker -->> RouteOrch: <br>
+    deactivate RouteBulker
+  end
+
+  Note left of RouteOrch: On any pre-condition failed<br>RouteOrch is retrying to remove <br>the route on next iteration<br>In this case no response is written<br>
+
+  RouteOrch ->> RouteBulker: RouteBulker.flush()
+  activate RouteBulker
+  RouteBulker ->> ASIC_DB: sai_route_api->remove_route_entries()
+  activate ASIC_DB
+  ASIC_DB ->> SYNCD: sai_route_api->remove_route_entries()
+  activate SYNCD
+  SYNCD -->> ASIC_DB: sai_status_t *object_statuses
+  deactivate SYNCD
+  ASIC_DB -->> RouteBulker: sai_status_t *object_statuses
+  deactivate ASIC_DB
+
+  RouteBulker -->> RouteOrch: <br>
+  deactivate RouteBulker
+
+  Note left of RouteOrch: On any route deletion error<br>RouteOrch crashes orchagent.<br> No response is written<br>
+
+  loop For each status in RouteBulker
+      RouteOrch ->> ResponsePublisher: ResponseBulisher.publish()
+      activate ResponsePublisher
+      ResponsePublisher ->> APPL_STATE_DB: Delete from ROUTE_TABLE
+      activate APPL_STATE_DB
+      APPL_STATE_DB -->> ResponsePublisher: <br>
+      deactivate APPL_STATE_DB
+      Note right of RouteOrch: Publish the status and remove entry from APPL_STATE_DB
+      ResponsePublisher -->> RouteOrch: <br>
+      deactivate ResponsePublisher
+  end
+
+  deactivate RouteOrch
+```
+
+#### 7.4. FPMsyncd
+
+
+> The dplane_fpm_nl has the ability to read route netlink messages
+> from the underlying fpm implementation that can tell zebra
+> whether or not the route has been Offloaded/Failed or Trapped.
+> The end developer must send the data up the same socket that has
+> been created to listen for FPM messages from Zebra.  The data sent
+> must have a Frame Header with Version set to 1, Message Type set to 1
+> and an appropriate message Length.  The message data must contain
+> a RTM_NEWROUTE netlink message that sends the prefix and nexthops
+> associated with the route.  Finally rtm_flags must contain
+> RTM_F_OFFLOAD, RTM_F_TRAP and or RTM_F_OFFLOAD_FAILED to signify
+> what has happened to the route in the ASIC.
+
+A new class *RouteFeedbackChannel* in fpmsyncd is introduced to manage orchagent responses and send the corresponding FPM message to zebra.
+
+Zebra requires to send RTM_NEWROUTE back with RTM_F_OFFLOAD flag set once route is programmed in HW. FpmSyncd create an RTM_NEWROUTE message based off VRF, prefix and nexthops that where actually installed which are part of publish notification.
+
+The offload route status is relevant only for BGP received routes. In order to send back RTM_NEWROUTE it is required to encode "protocol" (rtmsg->rtm_protocol field) in APPL_DB which will be sent back as part of response:
+
+```
+127.0.0.1:6379> hgetall "ROUTE_TABLE:193.5.96.0/25"
+1) "nexthop"
```

Comment:
Why dont we have 'vrf' as the key in the ROUTE_TABLE? we could be supporting only 'default' VRF now, but it's important to have the VRF as the key for future enhancements with data VRF.